src/app/(dashboard)/invoices/page.tsx

```diff
@@ -0,0 +1,421 @@
+'use client';
+
+import React, { useState } from 'react';
+import ItemsTable from './itemsList';
+import { env } from 'process';
+import Image from 'next/image';
+import { createSupbaseClient } from '@/lib/supabase/client';
+interface IReceipts {
+	id: string;
+	proj_id: string;
+	org_id: string;
+	img_id: string;
+	store: string;
+	category: string;
+	updated_by: string;
+	updated_at: Date;
+	created_by: string;
+	created_at: Date;
+	price_total: number;
+	note?: string;
+}
+
+const ReceiptPage = () => {
+	const DEFAULT_IMAGE =
+		'https://apqmqmysgnkmkyesdrnn.supabase.co/storage/v1/object/public/profile-avatars/wyncoservices.png';
+	const [receiptFile, setReceiptFile] = useState<File | null>(null);
+	const [receiptData, setReceiptData] = useState(null);
+	const [loading, setLoading] = useState(false);
+	const [error, setError] = useState('');
+	const [items, setItems] = useState([]);
+	const [image, setImage] = useState<any>([]);
+	const [imageURL, setImageURL] = useState(DEFAULT_IMAGE);
+	//FIXME: EXTRACT PROJECT_ID AMD org_id dyncmically
+
+	const [reciept, setReciept] = useState<IReceipts | null | any>({
+		proj_id: 'a7188d51-4ea8-492e-9277-7989551a3b97',
+		org_id: 'a210d3f7-bcc8-4e8b-9d61-2d4228bff047',
+		img_id: 'some-img-id',
+		store: 'store',
+		category: 'category',
+		updated_at: new Date(),
+		created_by: 'cd1d7916-a61a-40fa-9691-c29e42c8988a',
+		created_at: new Date(),
+		price_total: 100.0,
+		note: '',
+	});
+
+	const handleChange = (
+		event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement | any>,
+	) => {
+		const { name, value } = event.target;
+
+		// Create a new object for the state update
+		setReciept((prevReciept: any) => ({
+			...prevReciept, // Spread the previous state
+			[name]: name === 'price_total' ? parseFloat(value) : value, // Ensure numbers are handled correctly
+		}));
+	};
+
+	const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+		if (!event.target.files) {
+			return;
+		}
+
+		setImage(event.target.files[0]);
+		setImageURL(URL.createObjectURL(event.target.files[0]));
+		setReceiptFile(event.target.files ? event.target.files[0] : null);
+	};
+
+	const handleSubmit = (event: React.FormEvent) => {
+		event.preventDefault();
+		if (!receiptFile) {
+			setError('Please select a file to upload.');
+			return;
+		}
+
+		setLoading(true);
+		const form = new FormData();
+		form.append('file', receiptFile);
+		form.append('refresh', 'false');
+		form.append('incognito', 'false');
+		form.append('extractTime', 'false');
+		form.append('extractLineItems', 'true');
+
+		const options = {
+			method: 'POST',
+			headers: {
+				accept: 'application/json',
+				apikey: 'f0d78cd0be8511eeb72409b0b60cbbed',
```

Comment:
not suer if this should be public?!?!

Reply:
![UHM](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExdGxyN3preDBtY3dudGdpaG5nOW16M2M4enlocWVibWFkNWtpajc2NCZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/DhiRqIsofVMi7fWNBQ/giphy.gif)

Reply:
yeah I am aware the issue was that when ever I used to pull from process.env it would cause an error until I realized that its a client component, any tips for fixing ?

Reply:
![](https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExaG8xOHhycGpwamYzd3QzNmdzZHZ1enR4MHhveWJkemU2Mmg5NWt4eSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/TdWQV0lWuRVyR1JxuY/giphy.gif)